### ISSUE TITLE
Problems with subscriber in Android

### ISSUE THREAD

**Issue by joelpiccoli** [opentok/opentok-react-native#239]
Hello everyone, I'm trying to implement the OpenTok Lib for react-native but I'm facing some issues that I can't figured it out yet.
First one, and I think the most important, is that I can't connect as a Subscriber to a session without crashing the App, that occurs only on android devices, and for that I can't see if it's working the connections between two phones.

What am I doing wrong?

**Current behavior**
> The Android App crashes as soon as you open a session as a Subscriber.

**Steps to reproduce**
> Open a session as subscriber on a android device

**Example Project**

***Appointment***
```
import React, { Component } from 'react';
import { OTSession } from 'opentok-react-native';

import { TokBox } from '../../../helpers/constants';
import PublisherStream from './PublisherStream';
import SubscriberStream from './SubscriberStream';

class Appointment extends Component {
  static navigationOptions = {
    header: null
  };

  render() {
    const { navigation: { navigate, state: { params } } } = this.props;
    return (
      <OTSession style={{ flex: 1 }}
        apiKey={TokBox.API_KEY}
        sessionId={params.sessionId}
        token={params.token}
        eventHandlers={this.sessionEventHandlers}>
        <PublisherStream style={{ borderColor: 'red', borderWidth: 3, height: '50%' }} />
        <SubscriberStream style={{ borderColor: 'blue', borderWidth: 3, height: '50%' }} />
      </OTSession>
    );
  }
}

export default Appointment;
```
***PublisherStream***
```
import React, { Component } from 'react';
import { OTPublisher } from 'opentok-react-native';

class PublisherStream extends Component {
  publisherEventHandlers = {
    streamCreated: (event) => {
      console.log('Publisher stream created!', event);
    },
    streamDestroyed: (event) => {
      console.log('Publisher stream destroyed!', event);
    }
  };


  render() {
    return (
      <OTPublisher style={this.props.style} eventHandlers={this.publisherEventHandlers} />
    );
  }
}

export default PublisherStream;
```

***SubscriberStream***
```
import React, { Component } from 'react';
import { OTSubscriber } from 'opentok-react-native';

class SubscriberStream extends Component {
  subscriberProperties = {
    subscribeToAudio: true,
    subscribeToVideo: true,
  };

  subscriberEventHandler = {
    connected(event) {
      console.log('connected', event);
    },
    disconnected(event) {
      console.log('disconnected', event);
    },
    videoDataReceived(event) {
      console.log('videoDataReceived', event);
    },
    videoEnabled(event) {
      console.log('videoEnabled', event);
    },
    videoNetworkStats(event) {
      console.log('videoNetworkStats', event);
    }
  };

  render() {
    return (
      <OTSubscriber
        properties={this.subscriberProperties}
        eventHandlers={this.subscriberEventHandler}
        style={this.props.style}
      />
    );
  }
}

export default SubscriberStream;
```

***Package.json***
```
{
  {...omitted}
  "dependencies": {
    "opentok-react-native": "^0.9.5",
    "react": "^16.8.1",
    "react-native": "0.58.4",
  },
  {...omitted}
}

```
***Logcat logs***

```

02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: *********************** OPENTOK ANDROID SDK 2.15.3 ********************************
02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: **** Lib built on 20190117151249
02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: **** OpenTok Android SDK : d86d4fc8373d99ee992495bcd499cd0d0b077609
02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: **** Copyright 2018 TokBox, Inc.
02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: **** Licensed under the Apache License, Version 2.0
02-25 13:41:31.206 24225-24339/com.connect2me.app I/OpenTok Android SDK: ************************************************************************
02-25 13:41:31.284 24225-24339/com.connect2me.app D/JVM: JVM::Initialize@[tid=24339]
02-25 13:41:31.284 24225-24339/com.connect2me.app D/JVM: JVM::JVM@[tid=24339]
02-25 13:41:31.284 24225-24339/com.connect2me.app D/JVM: LoadClasses
02-25 13:41:31.284 24225-24339/com.connect2me.app D/JVM: name: org/webrtc/voiceengine/BuildInfo
02-25 13:41:31.285 24225-24339/com.connect2me.app D/JVM: name: org/webrtc/voiceengine/WebRtcAudioManager
02-25 13:41:31.285 24225-24339/com.connect2me.app D/JVM: name: org/webrtc/voiceengine/WebRtcAudioRecord
02-25 13:41:31.285 24225-24339/com.connect2me.app D/JVM: name: org/webrtc/voiceengine/WebRtcAudioTrack
02-25 13:41:31.433 24225-24339/com.connect2me.app D/AUDIO_FOCUS: Call State Changed
02-25 13:41:31.433 24225-24339/com.connect2me.app D/AUDIO_FOCUS: CALL_STATE_IDLE
02-25 13:41:31.452 24225-24339/com.connect2me.app I/VideoCaptureFactory: Using Camera Capturer
02-25 13:41:31.469 24225-24339/com.connect2me.app I/VideoCaptureFactory: Using GLSurfaceViews for render
02-25 13:41:35.391 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile 2130706433 for video/avc
02-25 13:41:35.408 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile/level 0/3 for video/mpeg2
02-25 13:41:35.418 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile/level 0/0 for video/mpeg2
02-25 13:41:35.418 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile/level 0/2 for video/mpeg2
02-25 13:41:35.418 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile/level 0/3 for video/mpeg2
02-25 13:41:35.445 24225-24225/com.connect2me.app I/VideoCapabilities: Unsupported profile 4 for video/mp4v-es
02-25 13:41:35.451 24225-24225/com.connect2me.app W/VideoCapabilities: Unrecognized profile 2130706433 for video/avc
02-25 13:41:35.517 24225-25654/com.connect2me.app W/VideoCapabilities: Unrecognized profile 2130706433 for video/avc
02-25 13:41:36.312 24225-25665/com.connect2me.app D/opentok-nativerenderer: VideoRenderOpenGles20: id 0
02-25 13:41:36.315 24225-25665/com.connect2me.app D/opentok-nativerenderer: Setup: number of textures 16, size 4096
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410] JNI DETECTED ERROR IN APPLICATION: use of invalid jobject 0x5dc
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410] "Thread-9008" prio=5 tid=70 Runnable
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   | group="main" sCount=0 dsCount=0 obj=0x134e22e0 self=0xb9c9ead8
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   | sysTid=25654 nice=0 cgrp=default sched=0/0 handle=0x9ddb6930
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   | state=R schedstat=( 311328130 53011927 258 ) utm=29 stm=2 core=3 HZ=100
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   | stack=0x9dcba000-0x9dcbc000 stackSize=1014KB
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   | held mutexes= "mutator lock"(shared held)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #00 pc 0035c715  /system/lib/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiPKcPNS_9ArtMethodEPv+116)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #01 pc 0033d80f  /system/lib/libart.so (_ZNK3art6Thread4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+138)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #02 pc 0024f6e1  /system/lib/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+760)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #03 pc 0024fde9  /system/lib/libart.so (_ZN3art9JavaVMExt9JniAbortFEPKcS2_z+68)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #04 pc 0033e36b  /system/lib/libart.so (_ZNK3art6Thread13DecodeJObjectEP8_jobject+174)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #05 pc 0031afcb  /system/lib/libart.so (_ZN3art35InvokeVirtualOrInterfaceWithVarArgsERKNS_33ScopedObjectAccessAlreadyRunnableEP8_jobjectP10_jmethodIDSt9__va_list+250)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #06 pc 0026bee7  /system/lib/libart.so (_ZN3art3JNI15CallVoidMethodVEP7_JNIEnvP8_jobjectP10_jmethodIDSt9__va_list+330)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #07 pc 0010f20d  /system/lib/libart.so (_ZN3art8CheckJNI11CallMethodVEPKcP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDSt9__va_listNS_9Primitive4TypeENS_10InvokeTypeE+892)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #08 pc 00110289  /system/lib/libart.so (_ZN3art8CheckJNI14CallVoidMethodEP7_JNIEnvP8_jobjectP10_jmethodIDz+40)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #09 pc 0006bb70  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #10 pc 000a1ab0  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #11 pc 000ada64  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #12 pc 000c2810  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #13 pc 000c79dc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #14 pc 000dcffc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #15 pc 000dbff8  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #16 pc 000dab34  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #17 pc 001133e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #18 pc 000d9b38  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #19 pc 000ba254  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #20 pc 0010f8b4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #21 pc 00104f00  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (uv_run+304)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #22 pc 000bb114  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #23 pc 000845e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #24 pc 00041b1b  /system/lib/libc.so (_ZL15__pthread_startPv+30)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   native: #25 pc 00019405  /system/lib/libc.so (__start_thread+6)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410]   (no managed stack frames)
02-25 13:41:36.508 24225-25654/com.connect2me.app A/art: art/runtime/java_vm_ext.cc:410] 
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] Runtime aborting...
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] Aborting thread:
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "Thread-9008" prio=5 tid=70 Native
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=0 dsCount=0 obj=0x134e22e0 self=0xb9c9ead8
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=25654 nice=0 cgrp=default sched=0/0 handle=0x9ddb6930
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=R schedstat=( 357490944 60374267 292 ) utm=33 stm=2 core=1 HZ=100
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0x9dcba000-0x9dcbc000 stackSize=1014KB
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes= "abort lock"
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 0035c715  /system/lib/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiPKcPNS_9ArtMethodEPv+116)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 0033d80f  /system/lib/libart.so (_ZNK3art6Thread4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+138)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 00321915  /system/lib/libart.so (_ZNK3art10AbortState10DumpThreadERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEPNS_6ThreadE+20)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 00321b89  /system/lib/libart.so (_ZN3art7Runtime5AbortEv+532)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 000f3d4d  /system/lib/libart.so (_ZN3art10LogMessageD2Ev+2212)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 0024f9dd  /system/lib/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1524)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #06 pc 0024fde9  /system/lib/libart.so (_ZN3art9JavaVMExt9JniAbortFEPKcS2_z+68)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #07 pc 0033e36b  /system/lib/libart.so (_ZNK3art6Thread13DecodeJObjectEP8_jobject+174)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #08 pc 0031afcb  /system/lib/libart.so (_ZN3art35InvokeVirtualOrInterfaceWithVarArgsERKNS_33ScopedObjectAccessAlreadyRunnableEP8_jobjectP10_jmethodIDSt9__va_list+250)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #09 pc 0026bee7  /system/lib/libart.so (_ZN3art3JNI15CallVoidMethodVEP7_JNIEnvP8_jobjectP10_jmethodIDSt9__va_list+330)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #10 pc 0010f20d  /system/lib/libart.so (_ZN3art8CheckJNI11CallMethodVEPKcP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDSt9__va_listNS_9Primitive4TypeENS_10InvokeTypeE+892)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #11 pc 00110289  /system/lib/libart.so (_ZN3art8CheckJNI14CallVoidMethodEP7_JNIEnvP8_jobjectP10_jmethodIDz+40)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #12 pc 0006bb70  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #13 pc 000a1ab0  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #14 pc 000ada64  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #15 pc 000c2810  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #16 pc 000c79dc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #17 pc 000dcffc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #18 pc 000dbff8  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #19 pc 000dab34  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.592 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #20 pc 001133e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #21 pc 000d9b38  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #22 pc 000ba254  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #23 pc 0010f8b4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #24 pc 00104f00  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (uv_run+304)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #25 pc 000bb114  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #26 pc 000845e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #27 pc 00041b1b  /system/lib/libc.so (_ZL15__pthread_startPv+30)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #28 pc 00019405  /system/lib/libc.so (__start_thread+6)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   (no managed stack frames)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] Dumping all threads without appropriate locks held: thread list lock mutator lock
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] All threads:
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] DALVIK THREADS (71):
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "Thread-9008" prio=5 tid=70 Runnable
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=0 dsCount=0 obj=0x134e22e0 self=0xb9c9ead8
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=25654 nice=0 cgrp=default sched=0/0 handle=0x9ddb6930
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=R schedstat=( 401716151 69270413 322 ) utm=37 stm=3 core=3 HZ=100
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0x9dcba000-0x9dcbc000 stackSize=1014KB
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes= "abort lock" "mutator lock"(shared held)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 0035c715  /system/lib/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiPKcPNS_9ArtMethodEPv+116)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 0033d80f  /system/lib/libart.so (_ZNK3art6Thread4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+138)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 00346edb  /system/lib/libart.so (_ZN3art14DumpCheckpoint3RunEPNS_6ThreadE+422)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 00347a23  /system/lib/libart.so (_ZN3art10ThreadList13RunCheckpointEPNS_7ClosureE+194)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 00347f49  /system/lib/libart.so (_ZN3art10ThreadList4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+124)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 00321b0b  /system/lib/libart.so (_ZN3art7Runtime5AbortEv+406)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #06 pc 000f3d4d  /system/lib/libart.so (_ZN3art10LogMessageD2Ev+2212)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #07 pc 0024f9dd  /system/lib/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1524)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #08 pc 0024fde9  /system/lib/libart.so (_ZN3art9JavaVMExt9JniAbortFEPKcS2_z+68)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #09 pc 0033e36b  /system/lib/libart.so (_ZNK3art6Thread13DecodeJObjectEP8_jobject+174)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #10 pc 0031afcb  /system/lib/libart.so (_ZN3art35InvokeVirtualOrInterfaceWithVarArgsERKNS_33ScopedObjectAccessAlreadyRunnableEP8_jobjectP10_jmethodIDSt9__va_list+250)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #11 pc 0026bee7  /system/lib/libart.so (_ZN3art3JNI15CallVoidMethodVEP7_JNIEnvP8_jobjectP10_jmethodIDSt9__va_list+330)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #12 pc 0010f20d  /system/lib/libart.so (_ZN3art8CheckJNI11CallMethodVEPKcP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDSt9__va_listNS_9Primitive4TypeENS_10InvokeTypeE+892)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #13 pc 00110289  /system/lib/libart.so (_ZN3art8CheckJNI14CallVoidMethodEP7_JNIEnvP8_jobjectP10_jmethodIDz+40)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #14 pc 0006bb70  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #15 pc 000a1ab0  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #16 pc 000ada64  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #17 pc 000c2810  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #18 pc 000c79dc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #19 pc 000dcffc  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #20 pc 000dbff8  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #21 pc 000dab34  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #22 pc 001133e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #23 pc 000d9b38  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #24 pc 000ba254  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #25 pc 0010f8b4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #26 pc 00104f00  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (uv_run+304)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #27 pc 000bb114  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #28 pc 000845e4  /data/app/com.connect2me.app-2/lib/arm/libopentok.so (???)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #29 pc 00041b1b  /system/lib/libc.so (_ZL15__pthread_startPv+30)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #30 pc 00019405  /system/lib/libc.so (__start_thread+6)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   (no managed stack frames)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "main" prio=5 tid=1 Native
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x7347a2a0 self=0xb86d8c88
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24225 nice=-4 cgrp=default sched=0/0 handle=0xb6fabb34
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 6995013045 963619877 10493 ) utm=592 stm=107 core=1 HZ=100
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xbe489000-0xbe48b000 stackSize=8MB
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00016a6c  /system/lib/libc.so (syscall+28)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 000f60bb  /system/lib/libart.so (_ZN3art17ConditionVariable4WaitEPNS_6ThreadE+82)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 003e4a23  /system/lib/libart.so (_ZN3artL12GoToRunnableEPNS_6ThreadE+502)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 000e56dd  /system/lib/libart.so (_ZN3art12JniMethodEndEjPNS_6ThreadE+8)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 00dc7835  /system/framework/arm/boot.oat (Java_android_os_BinderProxy_transactNative__ILandroid_os_Parcel_2Landroid_os_Parcel_2I+152)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.os.BinderProxy.transactNative(Native method)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.os.BinderProxy.transact(Binder.java:503)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.hardware.display.IDisplayManager$Stub$Proxy.getDisplayInfo(IDisplayManager.java:254)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.hardware.display.DisplayManagerGlobal.getDisplayInfo(DisplayManagerGlobal.java:119)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - locked <0x0b170510> (a java.lang.Object)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.view.Display.updateDisplayInfoLocked(Display.java:816)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.view.Display.getRotation(Display.java:607)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - locked <0x06b83809> (a android.view.Display)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at com.opentok.android.DefaultVideoCapturer.onPreviewFrame(DefaultVideoCapturer.java:370)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.hardware.Camera$EventHandler.handleMessage(Camera.java:1129)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.os.Handler.dispatchMessage(Handler.java:102)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.os.Looper.loop(Looper.java:148)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at android.app.ActivityThread.main(ActivityThread.java:5443)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.reflect.Method.invoke!(Native method)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "Signal Catcher" prio=5 tid=2 WaitingInMainSignalCatcherLoop
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c910a0 self=0xb9656c68
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24230 nice=0 cgrp=default sched=0/0 handle=0xb4591930
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 1140208 3260469 3 ) utm=0 stm=0 core=2 HZ=100
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xb4495000-0xb4497000 stackSize=1014KB
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00043590  /system/lib/libc.so (__rt_sigtimedwait+12)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 0001c227  /system/lib/libc.so (sigwait+22)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 00328581  /system/lib/libart.so (_ZN3art13SignalCatcher13WaitForSignalEPNS_6ThreadERNS_9SignalSetE+60)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 003296bf  /system/lib/libart.so (_ZN3art13SignalCatcher3RunEPv+210)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 00041b1b  /system/lib/libc.so (_ZL15__pthread_startPv+30)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 00019405  /system/lib/libc.so (__start_thread+6)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   (no managed stack frames)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "JDWP" prio=5 tid=3 WaitingInMainDebuggerLoop
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c950a0 self=0xb9657268
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24231 nice=0 cgrp=default sched=0/0 handle=0xb448d930
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 3008698 491199 16 ) utm=0 stm=0 core=1 HZ=100
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xb4391000-0xb4393000 stackSize=1014KB
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00043d40  /system/lib/libc.so (__pselect6+20)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 0001b575  /system/lib/libc.so (select+60)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 003ea4a7  /system/lib/libart.so (_ZN3art4JDWP12JdwpAdbState15ProcessIncomingEv+202)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 0025b25b  /system/lib/libart.so (_ZN3art4JDWP9JdwpState3RunEv+298)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 0025c0c7  /system/lib/libart.so (_ZN3art4JDWPL15StartJdwpThreadEPv+10)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 00041b1b  /system/lib/libc.so (_ZL15__pthread_startPv+30)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #06 pc 00019405  /system/lib/libc.so (__start_thread+6)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   (no managed stack frames)
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "FinalizerDaemon" prio=5 tid=4 Waiting
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c8a8e0 self=0xb96598d0
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24233 nice=0 cgrp=default sched=0/0 handle=0xb4280930
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 129162566 50897906 363 ) utm=9 stm=3 core=1 HZ=100
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xb417e000-0xb4180000 stackSize=1038KB
02-25 13:41:38.593 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00016a6c  /system/lib/libc.so (syscall+28)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 000f60bb  /system/lib/libart.so (_ZN3art17ConditionVariable4WaitEPNS_6ThreadE+82)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 002b04bf  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadExibNS_11ThreadStateE+1146)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 002b1175  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadEPNS_6mirror6ObjectExibNS_11ThreadStateE+112)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 002c228d  /system/lib/libart.so (_ZN3artL13Object_waitJIEP7_JNIEnvP8_jobjectxi+36)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 00000585  /system/framework/arm/boot.oat (Java_java_lang_Object_wait__JI+96)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Object.wait!(Native method)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - waiting on <0x0d78ca0e> (a java.lang.ref.ReferenceQueue)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Object.wait(Object.java:423)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:101)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - locked <0x0d78ca0e> (a java.lang.ref.ReferenceQueue)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:72)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:185)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.run(Thread.java:818)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "ReferenceQueueDaemon" prio=5 tid=5 Waiting
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c8a880 self=0xb96562b8
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24232 nice=0 cgrp=default sched=0/0 handle=0xb4387930
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 65195517 35359166 396 ) utm=3 stm=3 core=0 HZ=100
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xb4285000-0xb4287000 stackSize=1038KB
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00016a6c  /system/lib/libc.so (syscall+28)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 000f60bb  /system/lib/libart.so (_ZN3art17ConditionVariable4WaitEPNS_6ThreadE+82)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 002b04bf  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadExibNS_11ThreadStateE+1146)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 002b1175  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadEPNS_6mirror6ObjectExibNS_11ThreadStateE+112)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 002c2261  /system/lib/libart.so (_ZN3artL11Object_waitEP7_JNIEnvP8_jobject+32)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #05 pc 00000387  /system/framework/arm/boot.oat (Java_java_lang_Object_wait__+74)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Object.wait!(Native method)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - waiting on <0x0ed27c2f> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Daemons$ReferenceQueueDaemon.run(Daemons.java:147)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - locked <0x0ed27c2f> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.run(Thread.java:818)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "FinalizerWatchdogDaemon" prio=5 tid=6 Sleeping
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c8a940 self=0xb9659d40
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24234 nice=0 cgrp=default sched=0/0 handle=0xb4179930
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 1129425 6067760 19 ) utm=0 stm=0 core=0 HZ=100
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xb4077000-0xb4079000 stackSize=1038KB
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00016a70  /system/lib/libc.so (syscall+32)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #01 pc 000f6319  /system/lib/libart.so (_ZN3art17ConditionVariable9TimedWaitEPNS_6ThreadExi+104)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #02 pc 002b01fd  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadExibNS_11ThreadStateE+440)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #03 pc 002b1175  /system/lib/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadEPNS_6mirror6ObjectExibNS_11ThreadStateE+112)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #04 pc 0001c773  /system/framework/arm/boot.oat (Java_java_lang_Thread_sleep__Ljava_lang_Object_2JI+126)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.sleep!(Native method)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - sleeping on <0x0ccf623c> (a java.lang.Object)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.sleep(Thread.java:1031)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   - locked <0x0ccf623c> (a java.lang.Object)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.sleep(Thread.java:985)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Daemons$FinalizerWatchdogDaemon.sleepFor(Daemons.java:273)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Daemons$FinalizerWatchdogDaemon.waitForFinalization(Daemons.java:284)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Daemons$FinalizerWatchdogDaemon.run(Daemons.java:232)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   at java.lang.Thread.run(Thread.java:818)
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] 
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399] "HeapTaskDaemon" prio=5 tid=7 Blocked
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | group="" sCount=1 dsCount=0 obj=0x12c8a9a0 self=0xb965a1b0
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | sysTid=24235 nice=0 cgrp=default sched=0/0 handle=0xab074930
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | state=S schedstat=( 1435785515 201710271 1050 ) utm=127 stm=16 core=1 HZ=100
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | stack=0xaaf72000-0xaaf74000 stackSize=1038KB
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   | held mutexes=
02-25 13:41:38.594 24225-25654/com.connect2me.app A/art: art/runtime/runtime.cc:399]   native: #00 pc 00016a70  /system/lib/libc.so (syscall+32)
```

**What is the expected *correct* behavior?**
> The app should connect the two sides and share video correctly.

**Comment by msach22** [comment]
@joelpiccoli Thanks for filing the issue - sorry to hear that your application is crashing. Could you please share the Android version and device that you're using? I'd be happy to try and reproduce this issue and get this resolved for you.

**Comment by joelpiccoli** [comment]
hello @msach22 ! Thank you for your help!

```
ext {
        googlePlayServicesVersion = '15.0.0'
        buildToolsVersion = "27.0.3"
        minSdkVersion = 21
        compileSdkVersion = 27
        targetSdkVersion = 26
        supportLibVersion = "27.1.1"
    }
```

I'm testing the App on the device running Android 6.0.

**Comment by msach22** [comment]
Thanks @joelpiccoli  - I just ran this on a simulator running Android 6.0 and I'm not observing a crash. Is there a specific device that you're using?

You may have already done this, but could you please clean your build and build again?

**Comment by joelpiccoli** [comment]
> Thanks @joelpiccoli - I just ran this on a simulator running Android 6.0 and I'm not observing a crash. Is there a specific device that you're using?
> 
> You may have already done this, but could you please clean your build and build again?

I've already did the clean process, I'm using a Moto G3 to test the application, but the same happens in emulator with android Oreo too...
I don't what's happening... ;/

**Comment by joelpiccoli** [comment]
@msach22 I've created a new project, configured everything again and the subscriber keep crashing the app...

**Comment by msach22** [comment]
@joelpiccoli Please email me at manik[at]tokbox.com so we can set up a 1:1 debug session :)

**Comment by joelpiccoli** [comment]
sure @msach22! Thanks a lot for the help! :D

**Comment by joelpiccoli** [comment]
Actually, the problem was with a token that I was misconfusing the actually usage. Thanks a lot for @msach22 help and support!

**Comment by msach22** [comment]
Happy to help @joelpiccoli! Glad to hear that you're using the library!

**Comment by barmanrajdeep** [comment]
@joelpiccoli what was the wrong usage of the token?

**Comment by barmanrajdeep** [comment]
I was facing crashes too only in release builds. Got mine resolved later though. In my case it was related to important proguard rules for this library classes. The official docs should mention it, IMO

**Comment by joelpiccoli** [comment]
Glad to hear @barmanrajdeep ! I think the docs are in this repo, you could open a PR to add this to the docs!
